frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -401,31 +425,50 @@ function MainRender(props: {
+                const dataOffsetEnd = offset;
+
                 sliceElements.push(
                     <Grid
                         row
                         key={`slice:${currentReceiver}|${timeSlotStart}`}
-                        className={`slice ${sliceClassName}`}
-                        data-keyoffset={keyOffset - 1}
+                        className={`slice ${sliceClassName} ${sliceFilterClassName}`}
+                        data-offset={dataOffset}
+                        data-offset-end={dataOffsetEnd}
                         onClick={
                             successForSlice.currentState ===
                             SuccessRate.UNDEFINED
                                 ? undefined // do not even install a click handler noop
                                 : (evt) => {
-                                      // get saved offset from "data-keyoffset" attribute on this element
-                                      const dataKeyOffset = parseInt(
-                                          evt.currentTarget?.dataset[
-                                              "keyoffset"
-                                          ] || "-1"
+                                      // get saved offset from "data-offset" attribute on this element
+                                      const target = evt.currentTarget;
+                                      const sliceStart = parseInt(
+                                          target?.dataset["offset"] || "-1"
+                                      );
+                                      let sliceEnd = parseInt(
+                                          target?.dataset["offsetEnd"] || "-1"
                                       );
                                       // sanity check it's within range (should never happen)
                                       if (
-                                          dataKeyOffset >= 0 &&
-                                          dataKeyOffset < keys.length
+                                          sliceStart >= 0 &&
+                                          sliceStart < props.data.length
                                       ) {
-                                          const key = keys[dataKeyOffset];
-                                          onClick(props.data[key]);
+                                          // should never happen, but safety
+                                          sliceEnd =
+                                              sliceEnd > sliceStart
+                                                  ? sliceEnd
+                                                  : sliceStart;
+                                          const subsetData = props.data.slice(
+                                              sliceStart,
+                                              sliceEnd
+                                          );
+                                          props.onClick(subsetData);
```

Comment:
a note that this handler is generally going to be used to open the modal and pass data into it. Kind of hard to trace that flow through

Reply:
I agree that confusing since it's drilled down. I wasn't sure of a good solution since the modal UI lives at the top-most level. 

In a non-react-nested-component systems, we might have some messaging pipeline and dialog would be a listener?

Any suggestions on how to improve it?